### PR TITLE
cmd/update-report: check for HOMEBREW_DISABLE_LOAD_FORMULA

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This occasionally caused bottle jobs to fail.

### CODE REVIEW

File: Library/Homebrew/cmd/update-report.rb [hunk 1]
```diff
@@ -115,11 +115,13 @@ def update_report
         puts "No changes to formulae."
       else
         hub.dump
-        hub.reporters.each(&:migrate_tap_migration)
-        hub.reporters.each(&:migrate_formula_rename)
-        CacheStoreDatabase.use(:descriptions) do |db|
-          DescriptionCacheStore.new(db)
-                               .update_from_report!(hub)
+        unless ENV["HOMEBREW_DISABLE_LOAD_FORMULA"]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb]: Thoughts on doing an early return here instead?

File: Library/Homebrew/cmd/update-report.rb [hunk 2]
```diff
@@ -214,16 +216,18 @@ def report
...
-          formula = Formulary.factory(tap.path/src)
-          new_version = formula.pkg_version
-          old_version = FormulaVersions.new(formula).formula_at_revision(@initial_revision, &:pkg_version)
-          next if new_version == old_version
-        rescue FormulaUnavailableError
-          # Don't care if the formula isn't available right now.
-          nil
-        rescue Exception => e # rubocop:disable Lint/RescueException
-          onoe "#{e.message}\n#{e.backtrace.join "\n"}" if ARGV.homebrew_developer?
+        unless ENV["HOMEBREW_DISABLE_LOAD_FORMULA"]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/update-report.rb]: Thoughts on doing an early return here instead?

File: Library/Homebrew/cmd/update-report.rb (hunk 1 above)
Comment (Bo98) [reply Library/Homebrew/cmd/update-report.rb]: Could doing so and thus skip the `Homebrew.failed = true` line be a concern?

File: Library/Homebrew/cmd/update-report.rb (hunk 2 above)
Comment (Bo98) [reply Library/Homebrew/cmd/update-report.rb]: We could, yeah. I suppose skipping printing the new/changed/deleted formula list entirely doesn't really matter.

File: Library/Homebrew/cmd/update-report.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/update-report.rb]: Good catch. That env is set by `update.sh` so could be moved before the early return.

File: Library/Homebrew/cmd/update-report.rb (hunk 2 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/update-report.rb]: Aye 👍

**Comment by MikeMcQuaid** [comment]
Thanks again @Bo98!

**Pull request merged**